### DISCUSSION TITLE
Few Game Gear games can't be dumped fine

### DISCUSSION THREAD

**Discussion by PsyK0p4T** [sanni/cartreader#452] [Need help]
Hi everyone.

Recently I spent quite some time redumping Game Gear games with Retrode adapter. Also updated a bit the code to handle 32kb games. And I would say that around 95% of the games tested so far can be dumped just fine. The results match the current known roms, with strictly the same crc32, all good.

Exept a few titles. Of course all Codemasters games wich are using their own mappers, but let's keep them for later.
I'm more interested by other carts, which were manufactured by sega, and seem to react differently. Among them : Wimbledon, Super Monaco GP, G-LOC, Popils, Super Off Road (all were european cartridges, and all of them are 128kb) maybe other regions/games can react same...

**Classic behavior** : 
Use the slot 2 (0xFFFF) to address 16kb of rom data, read that slot, write data to file, and repeat this bank after bank, up to the full rom size.
Exept for 32kb games wich don't use mappers.

**Current behavior, with the games listed previously** : 
Slots 0-1-2 don't seem to be usable the same way to address rom banks.
The first 32kb of data (bank 0 + 1) can be read, without mapping slot 2, by looking at 0xFFFD and 0xFFFE.
But using slot 2 to address banks doesn't seem to work as for other titles.
Whatever bank is pushed to 0xFFFF, once read, that slot will only contain the very last bank. 
For exemple, with all the games listed previously, ROMs are 128kb, and the 7th bank is always the one readable at slot 2. Even if you try to map another bank to slot 2.

If someone is able to figure out what is happening here ?

Few more random info, maybe not useful but just in case... : 
- Popils got embedded sram onto its PCB. All other titles don't, and don't seem to have a any other special chip neither.
- I've tested different copies of "G-LOC" (eur version), to confirm that it was not related to a faulty cart. But no, both carts produced the same result.

**Comment by PsyK0p4T** [comment]
Adding some news here...

I purchased "Wimbledon", now the japanese cartridge. Tried to dump it, and everything worked fine, not any problem, I ended with the expected CRC.

Title in particular is interesting, cause the rom is the same all around the world. EUR/USA/JPN carts are all using the same rom.
Unfortunately, even if the cart PCB is the same model in EUR and JPN cartridges, the ROM chip seems different and do not react same. EUR version is still giving the problem detailed on first post :(

**Comment by Mythris29** [comment]
Just dumped G-Loc today using a Retron 3-in-1 adapter. It was successful and matched the correct CRC.

**Comment by PsyK0p4T** [comment]
Aww really... Can you please detail : 
- cartridge origin (eur/usa/jpn) ?
- which software version you used
- post here a picture of opened cartridge, so I can compare the chip you have in with mines.
Thanks a lot :)

**Reply by Mythris29** [reply]
I just hope this helps. 

USA/EUR/Brazil (at least that's what the CRC check says)
9.5 (with the 3-in-1 retron sms.ino)
Chip looks strange...not sure if it could be a repo or not I don't have as much experience with Game Gear games.

![PXL_20220830_162908198](https://user-images.githubusercontent.com/15204330/187491198-db8c5c01-b0dc-4e09-8752-4a4b0268c723.jpg)
![PXL_20220830_162917135](https://user-images.githubusercontent.com/15204330/187491204-46366d92-7c6c-49b9-bfec-dc315963ec30.jpg)
![PXL_20220830_163022922](https://user-images.githubusercontent.com/15204330/187491205-ea0f060e-c1ef-4e9e-b817-248a8947f94a.jpg)

**Comment by PsyK0p4T** [comment]
I don't think it's a repro. But it explains why it worked for you... our PCBs are different.
That's also what happened to me with Wimbledon, EUR and JPN carts had different chips.
Here are the G-LOC I've here : 

![front](https://user-images.githubusercontent.com/87064902/187791311-4810faf6-716a-4b3e-bca3-b8b8cc80777b.jpg)
![pcb1](https://user-images.githubusercontent.com/87064902/187791336-23933145-5717-41b3-8e3c-0ea899c969b6.jpg)
![pcb2](https://user-images.githubusercontent.com/87064902/187791353-bbfa806c-1ff8-475e-adb5-0588d70367a3.jpg)

**Comment by Mythris29** [comment]
As an update to this. I recently made a retrode SMS game gear adapter...and G.Loc was the only game out of the 3 I tried that didn't dump correctly. So there is something to this but I think it's mode/adapter specific. Hopefully that helps.

**Reply by Mythris29** [reply]
But the retron 3-in-1 is able to dump this cart successfully now repeatedly.

**Comment by PsyK0p4T** [comment]
Huh really ? so... it would not be the code fault...

Do you have the possibility to open the Retron adapter without breaking anything, and take big resolution pictures of the pcb (both sides) ? It would be interesting to compare what are the pins mapping differences between the two adapters.

Skaman recently reported that another Retrode adapter (Atari 2600) wasn't really mapped correctly. So I'm curious now.

**Reply by Mythris29** [reply]
[Photos-001 (3).zip](https://github.com/sanni/cartreader/files/9501227/Photos-001.3.zip)

Hopefully these are helpful

**Reply by PsyK0p4T** [reply]
Thanks a lot for the effort Mythris. Not easy to work with unfortunately. And that components added to the pcb... I was expecting something easier to follow. It would be easier for me to use a multimeter and test the conductivity of each pins at the cart slot, and to the final connector at bottom. I will try to find a retron adapter somewhere :/

**Comment by PsyK0p4T** [comment]
Digging up this thread.
As reported by Mythris, it looks like the Retron 3-in-1 adapter is giving better results than the Retrode one for Game Gear.
All the games listed in my initial post are dumped fine using a Retron 3-in-1 adapter. And no change was made to the code. So... could be the pins mapping ? or maybe the components inside Retron adapter ? Can't say. But for now I would recommend it for a better compatibility with the GG set.

Games that are not using SEGA official mappers (like Codemasters games, pirate carts etc) are still not compatible of course, and would need more dev.
Next step, out of curiosity : Dumping SMS carts using [memory paging chip](https://github.com/sanni/cartreader/issues/527) with the Retron adapter. Just to see the result...

**Comment by Ancyker** [comment]
Worked fine with my adapter :3

![2023-03-21 01 42 14](https://user-images.githubusercontent.com/7105333/226527451-fc55e7ca-6f26-4d9f-8411-7e18f3ea690e.jpg)

**Reply by PsyK0p4T** [reply]
Nice :) if you have Fantasy Zone, this one is also a good test. I've seen reports of both JPN and EUR carts failing to dump with retrode adapter (confirmed it here) but it worked fine with Retron 3-in-1.

Which mode are you selecting in OSCR menu with your adapter ?

**Reply by Ancyker** [reply]
@PsyK0p4T 3-in-1, it also works if you choose SMS.
